### PR TITLE
[action] Generate update_code_signing_settings action instead of deprecated automatic_code_signing

### PR DESCRIPTION
**Pull request by att55**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

`automatic_code_signing` has `use_automatic_signing`, the key switches `Auto` to `Manual` code signing. And the defaulut value is `false`.

https://docs.fastlane.tools/actions/automatic_code_signing/#automatic_code_signing

Sample code is as below.

```
automatic_code_signing(
  path: "demo-project/demo/demo.xcodeproj",
  use_automatic_signing: false
)

automatic_code_signing(
  path: "demo-project/demo/demo.xcodeproj",
  use_automatic_signing: true
)
```

But, I feel the specifications below are strange and confusing users. 

1. Even though the action name is `automatic_code_signing`, the action allows to switch `Auto` to `Manual` code signing.
2. Even though the action name is `automatic_code_signing`, `Manual` code signing is executed by default because the default value of `use_automatic_signing` is `false`.

So, I added `code_signing` as alias of `automatic_code_signing` to prevent confusing users.

### CODE REVIEW

**Review by joshdholtz — CHANGES_REQUESTED** [review]
@att55 👋 This is a very interesting change! I'm a little bit for but also a little bit against this 🤔 I do agree that this action should have a different name but I also don't want to just introduce another action/alias.

Aliases are nice (in this cases since some of the options conflict with the name of the action) but it makes things a little bit harder to maintain and harder to find other actions. 

If we are going to do this, I would rather move the logic from `AutomaticCodeSigning` into one that is better named and add a message saying that we are deprecating `AutomaticCodeSigning` in the future.

The `code_signing` name feels a bit too generic to me (I actually thought this PR was for something like _sigh_ or _cert_ at first). I think it would need to be something like `update_code_signing_settings`

Thoughts?

**Comment by att55** [comment]
@joshdholtz Thank you so much for your review!

> I would rather move the logic from AutomaticCodeSigning into one that is better named and add a message saying that we are deprecating AutomaticCodeSigning in the future.
> I think it would need to be something like update_code_signing_settings

I think your plan is very good and `update_code_signing_settings` is effective to resolve this iseue! 😸 
I will try to adopt your plan in this PR 👍

**Comment by att55** [comment]
@joshdholtz I adopted our plan in this PR 👍Please check when you can 🙏 

The outline is below.
- added the message saying that we are deprecating automatic_code_signing in automatic_code_signing action .
- generated update_code_signing_setttings action instead of automatic_code_signing
- added a test file for update_code_signing_setttings
- added fixtures(xcodeproj) to exec the test of update_code_signing_setttings

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 1]
```diff
@@ -34,6 +34,8 @@
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
       expect(UI).to receive(:success).with("\t * today")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 2]
```diff
@@ -34,6 +34,8 @@
...
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 3]
```diff
@@ -76,6 +78,8 @@
       # G3KGXDXQL9
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 4]
```diff
@@ -76,6 +78,8 @@
...
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 5]
```diff
@@ -106,6 +110,8 @@
       # G3KGXDXQL9
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 6]
```diff
@@ -106,6 +110,8 @@
...
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 7]
```diff
@@ -142,6 +148,8 @@
       # G3KGXDXQL9
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 8]
```diff
@@ -142,6 +148,8 @@
...
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 9]
```diff
@@ -174,6 +182,8 @@
       # G3KGXDXQL9
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 10]
```diff
@@ -174,6 +182,8 @@
...
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 11]
```diff
@@ -206,6 +216,8 @@
       # G3KGXDXQL9
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 12]
```diff
@@ -206,6 +216,8 @@
...
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 13]
```diff
@@ -236,6 +248,8 @@
 
     it "targets not found notice" do
       allow(UI).to receive(:important)
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

File: fastlane/spec/actions_specs/automatic_code_signing_spec.rb [hunk 14]
```diff
@@ -236,6 +248,8 @@
...
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
Comment (joshdholtz) [inline fastlane/spec/actions_specs/automatic_code_signing_spec.rb]:
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

**Review by joshdholtz — CHANGES_REQUESTED** [review]
Thanks SO much for making the changes I suggested 🥰 One more small change! If we could use `UI.deprecated` instead of `UI.important` that would be 💯

**Comment by att55** [comment]
@googlebot I consent.

**Comment by att55** [comment]
@googlebot I consent.

**Comment by att55** [comment]
@googlebot I consent.

**Comment by att55** [comment]
I thought I've already consented CLAs... 🤔??

**Comment by att55** [comment]
@joshdholtz Thank you so much for your review and requesting suggested changes to this PR!  
I fixed to use `UI.deprecated` instead of `UI.important`. 👍 
Sorry to bother you, but please check when you can 😄

**Comment by att55** [comment]
Sorry... I didn't understand how to pass "size-label" and "cla/google"...

**Comment by att55** [comment]
@joshdholtz Sorry to bother you... please check this PR when you can 🙏

**Comment by joshdholtz** [comment]
Yo yo yo! Will check first thing tomorrow morning ❤️

**Comment by att55** [comment]
Thank you so much ❤️ It's file, when you really have time 😸

**Comment by att55** [comment]
@joshdholtz ( When you can... 👀🙏 )

**Comment by joshdholtz** [comment]
Aye aye! Getting back to things first thing tomorrow morning 👌

**Comment by joshdholtz** [comment]
@googlebot I consent.

**Review by joshdholtz — APPROVED** [review]
Looks good to me! Thanks for making those changes 😊 Appreciate all your contributions ❤️

**Pull request merged**